### ISSUE TITLE
Removing warnings for undecorated 'new C()` and `--no-warnings` flags

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#10990]
After 1.18 is released and we've lived in the current world for some time, we should remove
the warnings for undecorated `new C()` calls (for class C) and also remove the --no-warnings
flags that were added to tests to squash these warnings.

Here is a (partial?) list of commits that added such `--no-warnings` flags:

* b7db99414866fb81916ee16fa97a251a654676de
* 45a8fa78477ba63b001e306439ca8e79b7d1b1f3
* 80debd50366e12b4c4df9d716deb404ba73892f3
* aa36cc493f88abdff3ba223c9461a13a1c8ba828
* b13c91330eac4099b36c02386783af20c604e26a
* 952b13302ef5bbd700e9a662ab1d95c66a343ff1
* 37d695c04e86c09b3f674db917d61904402ad7b4
* 61d735d4ac1359e8e4c65b1357f4c3c8c99163f3

**Comment by bradcray** [comment]
Tagging @ronawho on this for peace of mind as he asked about the `--no-warnings` added

**Comment by mppf** [comment]
@bradcray - another way to help remember is to add something to test/deprecated. In fact I already did that in test/deprecated/new-undecorated-warning.chpl. Of course this issue can help us remember to remove the --no-warnings on tests.

**Comment by bradcray** [comment]
> In fact I already did that in test/deprecated/new-undecorated-warning.chpl.

Yeah, I saw that and added one of my own as well.

> Of course this issue can help us remember to remove the --no-warnings on tests.

Right, that's exactly why I created it.  I didn't want to have to add a new SHA to the comments of a test in the repository every time I add a new `--no-warnings` flag (which I expect to be doing a fair amount of this week).  And I didn't want to crawl the git history to determine which commits introduced new `--no-warnings` flags.  Sorry if that wasn't obvious.
